pkg/api/documentsschema/documentsschema.proto

```diff
@@ -0,0 +1,252 @@
+syntax = "proto3";
+
+package immudb.documentsschema;
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/struct.proto";
+import "schema.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+option go_package = "github.com/codenotary/immudb/pkg/api/documentsschema";
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+  base_path: "/api/v2",
+  info: {
+    title: "immudb REST API v2";
+    description: "Objects storage API"
+  };
+  security_definitions: {
+    security: {
+      key: "bearer"
+      value: {
+        type: TYPE_API_KEY
+        in: IN_HEADER
+        name: "grpc-metadata-sessionid"
+        description: "Session token: <token>"
+      }
+    }
+  }
+  security: {
+    security_requirement: {
+      key: "bearer"
+    }
+  }
+};
+
+message DocumentInsertRequest {
+  string collection = 1;
+  repeated google.protobuf.Struct document = 2;
+}
+
+message DocumentInsertResponse {
+  schema.VerifiableTx proof = 1;
+}
+
+
+enum QueryOperator {
+  EQ = 0;
+  GT = 1;
+  GTE = 2;
+  LT = 3;
+  LTE = 4;
+  LIKE = 5;
+}
+
+message IndexValue {
+  oneof kind {
+    google.protobuf.NullValue null_value = 1;
```

Comment:
not sure I understand the need of `null_value`

Reply:
There is no way to distinguish between null and 0 in GRPC
If we want to allow null values for index fields - then we will need that, otherwise not

Reply:
so it's to be explicit on the null value then? is it required for searching for a document with a null value in a particular attribute?

Reply:
in fact, it's actually equivalent to what's defined for SQLValue